gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/version/HiveDatasetVersionCleaner.java

```diff
@@ -88,20 +88,24 @@ public void clean() throws IOException {
       Partition partition = hiveDatasetVersion.getPartition();
       try {
         if (!cleanableHiveDataset.isSimulate()) {
+          if (cleanableHiveDataset.isShouldDeleteData()) {
+            cleanableHiveDataset.getFsCleanableHelper().clean(hiveDatasetVersion, possiblyEmptyDirectories);
```

Comment:
are we certain this is an idempotent operation that won't throw an exception the second and more time?

scenario:
1. the first time through it was deleted before dropping of the partition failed
2. upon retry, before again trying to drop the partition, we invoke this once more to delete a FS location that was already removed

Reply:
I see the concern here. It always fetches the paths to be deleted from the `FileSystemDatasetVersion`. This can be confirmed with this line: `Set<Path> pathsToDelete = versionToDelete.getPaths();`. Thus, the set should be empty if we try to fetch paths that don't exist. However, upon re-run, we can also set the flag `isShouldDeleteData` to `false` which will not try to delete it again.